include/net/nrf_cloud_alerts.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_ALERTS_H_
+#define NRF_CLOUD_ALERTS_H_
+
+/** @file nrf_cloud_alerts.h
+ * @brief Module to provide mechanism to send application alerts to nRF Cloud.
+ */
+
+#include <zephyr/kernel.h>
+#include <math.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct nrf_cloud_rest_context;
+
+/** @brief Special value to pass when value should not be sent. */
+#define NRF_CLOUD_ALERT_UNUSED_VALUE NAN
+
+/** @defgroup nrf_cloud_alerts nRF Cloud Alerts
+ * @{
+ */
+
+/** @brief Alert type */
+enum nrf_cloud_alert_type {
+	/** Placeholder; not meant to be sent. */
+	ALERT_TYPE_NONE,
+	/** Device finished booting. */
+	ALERT_TYPE_DEVICE_NOW_ONLINE,
+	/** Device finished rebooting from a watchdog timeout. */
+	ALERT_TYPE_DEVICE_NOW_ONLINE_WDT,
+	/** Device is shutting down. */
+	ALERT_TYPE_DEVICE_GOING_OFFLINE,
+	/** Generic message (see description field). */
+	ALERT_TYPE_MSG,
+	/** Operating temperature outside of limits. Value is in degrees celcius.  */
+	ALERT_TYPE_TEMPERATURE,
+	/** Operating humidity outside of limits. Value is in percent relative humidity. */
+	ALERT_TYPE_HUMIDITY,
+	/** Battery level too low. Value is in volts. */
+	ALERT_TYPE_BATTERY,
+	/** Shock and vibration limits exceeded. Value is in m/s^2. */
+	ALERT_TYPE_SHOCK,
+	/** User specific alerts start here. */
+	ALERT_TYPE_CUSTOM = 100
+};
+
+/** @brief Alert data */
+struct nrf_cloud_alert_info {
+	/** The type of alert this represents. */
+	enum nrf_cloud_alert_type type;
+	/** A value associated with this alert. */
+	float value;
+	/** An optional human-readable description. */
+	const char *description;
+	/** A monotonically increasing alert counter used when current time not available. */
+	unsigned int sequence;
+	/** UNIX epoch timestamp (in milliseconds) of the data.
+	 * If a negative value is provided, the timestamp will be ignored.
+	 */
+	int64_t ts_ms;
+};
+
+/**
+ * @brief Transmit the specified alert to nRF Cloud.
+ *
+ * @param[in] type What kind of alert this is.
+ * @param[in] value Optional numeric value associated with the alert. If set to
+ *                  NRF_CLOUD_ALERT_UNUSED_VALUE, then do not send this field.
+ * @param[in] description Optional text describing the alert.
+ *
+ * @return 0 if alert successfully sent, else negative error number on error.
+ */
+int nrf_cloud_alert_send(enum nrf_cloud_alert_type type, float value, const char *description);
+
+/**
+ * @brief Transmit the specified alert to nRF Cloud.
```

Comment:
So this is identical to the previous description. This must be for the REST communication.